src/components/LoginButton.jsx

```diff
@@ -1,37 +1,59 @@
 import React, { useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
 import firebase from 'firebase';
+import { CircularProgress, makeStyles } from '@material-ui/core';
 import Button from '@material-ui/core/Button';
 import ProfileMenu from './ProfileDropdown';
 
+const useStyles = makeStyles(() => ({
+  progress: {
+    display: 'block',
+    margin: 'auto',
+  },
+}));
+
 export default function LoginButton() {
   const [isSignedIn, setIsSignedIn] = useState(!!firebase.auth().currentUser);
+  const [loading, setLoading] = useState(true);
+  const [error] = useState();
+  const classes = useStyles();
+
+  function updateUserStatus(user) {
+    setLoading(false);
+    setIsSignedIn(!!user);
+  }
 
   useEffect(
-    () => firebase.auth().onAuthStateChanged((user) => setIsSignedIn(!!user)),
+    () => firebase.auth().onAuthStateChanged((user) => updateUserStatus(user)),
     []
   );
 
   const signUserOut = () => firebase.auth().signOut();
   const user = firebase.auth().currentUser;
 
-  return isSignedIn ? (
-    <ProfileMenu
-      signOut={signUserOut}
-      name={user.displayName}
-      email={user.email}
-      photo={user.photoURL}
-    />
-  ) : (
-    <Button
-      variant="contained"
-      color="primary"
-      component={Link}
-      to={{
-        state: { showLoginDialog: true },
-      }}
-      replace>
-      Login
-    </Button>
+  return (
+    <>
```

Comment:
I think if you put a width here, it will prevent the other links to shift when the button changes to an Avatar. I'm not sure what width that would be set to though.

Reply:
Hm the width would be applied where? I included `width` and `marginLeft` to the avatar element and got it to align as close as possible to the link location when the login button is rendered

Reply:
in the `<>` although you would have to change it to `<div className={loginSection}>` and in your useStyles make the class and set the width. I checked in the inspector and I think the Login button is 64px in width. So that might work.